### DISCUSSION TITLE
Question about interaction with TDS

### DISCUSSION THREAD

**Discussion by Learwzr** [CURENT/andes#463] [Q&A] [answered]
Dear Dr.Cui,
I am trying to use a neural network controller in load frequency control. I want to know how to interacte with TDS or integrate a neural network controller in PV models. For example, the controller can get the frequency or ACE signals and modify the setpoints of generators every 0.1s during the TDS runing.
Best wishes! Looking forward to your answer!

**Comment by jinningwang** [comment, marked as answer]
This example discussed how to do the AGC power with ANDES, https://docs.andes.app/en/latest/_examples/ex8.html

**Reply by Learwzr** [reply]
I will try it. Thanks for your answer!

**Reply by longchangkai666-lab** [reply]
Hi there, I attempted to reopen this example file, yet it won’t open. Would you mind resending it?

**Comment by longchangkai666-lab** [comment]
Hello. I’m also researching the application of neural networks in frequency regulation. I’d like to know how you integrate agents into the power grid. Looking forward to your reply.
